akka-actor-typed/src/main/scala/akka/actor/typed/scaladsl/Behaviors.scala

```diff
@@ -33,30 +34,16 @@ object Behaviors {
   def setup[T](factory: ActorContext[T] ⇒ Behavior[T]): Behavior[T] =
     Behavior.DeferredBehavior(factory)
 
-  /**
-   * Factory for creating a [[MutableBehavior]] that typically holds mutable state as
-   * instance variables in the concrete [[MutableBehavior]] implementation class.
-   *
-   * Creation of the behavior instance is deferred, i.e. it is created via the `factory`
-   * function. The reason for the deferred creation is to avoid sharing the same instance in
-   * multiple actors, and to create a new instance when the actor is restarted.
-   *
-   * @param factory behavior factory that takes the child actor’s context as argument
-   * @return the deferred behavior
-   */
-  def mutable[T](factory: ActorContext[T] ⇒ MutableBehavior[T]): Behavior[T] =
-    setup(factory)
-
   /**
    * Mutable behavior can be implemented by extending this class and implement the
    * abstract method [[MutableBehavior#onMessage]] and optionally override
    * [[MutableBehavior#onSignal]].
    *
-   * Instances of this behavior should be created via [[Behaviors#Mutable]] and if
+   * Instances of this behavior should be created via [[Behaviors#setup]] and if
    * the [[ActorContext]] is needed it can be passed as a constructor parameter
    * from the factory function.
    *
-   * @see [[Behaviors#Mutable]]
+   * @see [[Behaviors#setup]]
    */
   abstract class MutableBehavior[T] extends ExtensibleBehavior[T] {
```

Comment:
not sure myself, but what do you think of moving `MutableBehavior` to a top level class instead of nested within `Behaviors`?

Reply:
vote for top level